src/components/DateRangePicker.jsx

```diff
@@ -138,7 +138,9 @@ export default class DateRangePicker extends React.Component {
 
   onEndDateChange(endDateString) {
     const { startDate, isOutsideRange, onDatesChange, onFocusChange, displayFormat } = this.props;
-    const endDate = toMomentObject(endDateString, displayFormat);
+
+    const formatString = typeof displayFormat === 'function' ? displayFormat() : displayFormat;
```

Comment:
the propTypes help ensure it's only a string or function, but perhaps it would be more robust (ie, fail faster when it's not a func or string) if this did `typeof displayFormat === 'string' ? displayFormat : displayFormat();`?